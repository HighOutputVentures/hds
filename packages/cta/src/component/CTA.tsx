import React, { FC, ReactElement } from 'react';
import {
  Box,
  Text,
  Flex,
  InputGroup,
  InputLeftElement,
  Input,
  Button,
} from '@chakra-ui/react';
import { EmailIcon } from '@chakra-ui/icons';

export interface CTAProps {
  header?: string;
  subHeader?: string;
  secondaryLabel?: string;
  primaryLabel?: string;
  onClose?: () => void;
  onSubmit?: () => void;
  content?: ReactElement;
  isSubscription?: boolean;
}

const CTA: FC<CTAProps> = ({
  header,
  subHeader,
  secondaryLabel,
  primaryLabel,
  onSubmit,
  onClose,
  isSubscription,
  content,
}) => (
  <Box>
    <Flex
      direction="column"
      w={{ base: 'auto', lg: '48rem' }}
      border="2px"
      borderColor="neutrals.200"
      borderRadius="0.5rem"
      justifyContent="center"
      p="24px"
    >
      <Text
        fontWeight={500}
        size={{ base: 'label-sm-default', lg: 'label-md-default' }}
      >
        {header}
      </Text>
      <Text
        color="neutrals.600"
        fontSize={14}
        lineHeight="20px"
        letterSpacing="0.02em"
      >
        {subHeader}
      </Text>

      <Box>{content}</Box>

      <Flex
        direction={{ base: 'column', lg: 'row' }}
        gap={{ base: 2, lg: 4 }}
        mt="1.25rem"
      >
        {isSubscription ? (
          <Flex direction="column" gap={2}>
            <Text size="label-xs-default" fontWeight={500}>
              Subscribe to updates
            </Text>
            <Flex
              direction={{ base: 'column', lg: 'row' }}
              gap={{ base: 2, lg: 4 }}
            >
              <InputGroup>
                <InputLeftElement
                  pointerEvents="none"
                  children={<EmailIcon color="neutrals.700" />}
                />
                <Input type="email" placeholder="you@untitledui.com" />
              </InputGroup>
              <Button
                onClick={onSubmit}
                px="2rem"
                width={{ base: '100%', lg: 'auto' }}
              >
                <Text size="button-default">{primaryLabel}</Text>
              </Button>
            </Flex>
          </Flex>
        ) : (
          <>
            {secondaryLabel && (
              <Button
                onClick={onClose}
                variant="outline"
                width={{ base: '100%', lg: 'auto' }}
              >
                <Text color="neutrals.700" size="button-default">
                  {secondaryLabel}
                </Text>
              </Button>
            )}
            {primaryLabel && (
              <Button onClick={onSubmit} width={{ base: '100%', lg: 'auto' }}>
                <Text size="button-default">{primaryLabel}</Text>
              </Button>
            )}
          </>
        )}
      </Flex>
    </Flex>
  </Box>
);

export default CTA;
