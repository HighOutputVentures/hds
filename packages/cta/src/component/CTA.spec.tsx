import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import React from 'react';
import CTA from './CTA';

describe('CTA', () => {
  it('should render CTA', () => {
    const { container } = render(
      <CTA
        header={`We've just released a new update!`}
        subHeader="Check out the all new dashboard view. Pages and now load faster."
        primaryLabel="Changelog"
        secondaryLabel="Dismiss"
      />
    );

    expect(container).toMatchSnapshot();
  });

  it('should have a header and sub header', () => {
    const { getByText } = render(
      <CTA
        header={`We've just released a new update!`}
        subHeader="Check out the all new dashboard view. Pages and now load faster."
        primaryLabel="Changelog"
        secondaryLabel="Dismiss"
      />
    );

    expect(getByText(`We've just released a new update!`)).toBeInTheDocument();
    expect(
      getByText(
        `Check out the all new dashboard view. Pages and now load faster.`
      )
    ).toBeInTheDocument();
  });
});
