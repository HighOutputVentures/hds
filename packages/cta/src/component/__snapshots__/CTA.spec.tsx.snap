// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CTA should render CTA 1`] = `
<div>
  <div
    class="css-0"
  >
    <div
      class="css-1vrjhbt"
    >
      <p
        class="chakra-text css-10rvbm3"
      >
        We've just released a new update!
      </p>
      <p
        class="chakra-text css-1tg9lzs"
      >
        Check out the all new dashboard view. Pages and now load faster.
      </p>
      <div
        class="css-0"
      />
      <div
        class="css-qqq897"
      >
        <button
          class="chakra-button css-1j8h89v"
          type="button"
        >
          <p
            class="chakra-text css-162yuu5"
          >
            Dismiss
          </p>
        </button>
        <button
          class="chakra-button css-1j8h89v"
          type="button"
        >
          <p
            class="chakra-text css-0"
          >
            Changelog
          </p>
        </button>
      </div>
    </div>
  </div>
</div>
`;
